backend/Squirrel.Core/Squirrel.Core.WebAPI/Controllers/ProjectController.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Squirrel.Core.Common.DTO.Projects;
+using Squirrel.Core.BLL.Interfaces;
+using Squirrel.Core.Common.Enums;
+
+namespace Squirrel.Core.WebAPI.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class ProjectController : ControllerBase
+    {
+        private readonly IProjectService _projectService;
+
+        public ProjectController(IProjectService projectService)
+        {
+            _projectService = projectService;
+        }
+
+        [HttpPost]
+        public async Task<ActionResult<ProjectDto>> AddProject(ProjectDto projectDto)
+        {
+            var addedProject = await _projectService.AddProjectAsync(projectDto);
+            
+            return Ok(addedProject);
+        }
+
+        [HttpPut("{projectId}")]
+        public async Task<ActionResult<ProjectDto>> UpdateProject(int projectId, ProjectDto projectDto)
+        {
+            var updatedProject = await _projectService.UpdateProjectAsync(projectId, projectDto);
+            if (updatedProject == null)
+            {
+                return NotFound();
+            }
```

Comment:
i created RequestException, that contains error message and HttpStatusCode, you can use it: create exception EntityNotFoundException (or something like this) and throw it in service or controller. GenericExceptionMiddleware will catch this exception and return a proper error message and status code